lib/topic_view.rb

```diff
@@ -604,8 +604,9 @@ def setup_filtered_posts
 
     if SiteSetting.ignore_user_enabled
       @filtered_posts = @filtered_posts.where.not("user_id IN (?) AND id <> ?",
-                                                  IgnoredUser.where(user_id: @user.id).select(:ignored_user_id),
+                                                  IgnoredUser.where(user_id: @user&.id).select(:ignored_user_id),
                                                   first_post_id)
+      @contains_gaps = true
```

Comment:
This one liner basically enables us to invoke the [Gaps](https://github.com/discourse/discourse/blob/03a70ef69c415d561dedcf3c5d453ce8944cd117/lib/topic_view.rb#L110) calculator, which basically will take a full set of posts for that topic and a subset of filtered post, and then it calculates where the gaps should be.